### PR TITLE
Code formatting, clean-up and docstring additions.

### PR DESCRIPTION
**Pull request by WilliamBerryiii**
This PR is for general code formatting, clean-up and docstring additions.  

Docstring formatting is done in consultation with the [Google Style Guide](http://google.github.io/styleguide/pyguide.html).  In some instances reStructuredText format has been moved to the Google style to improve readability and clarity, since documentation for this tool is not auto-generated. 
Other minor code cleanup includes adding intermediate variable before return statements in functions to assist with code debugging, intellisense, etc.

### CODE REVIEW

**Comment by YingXue** [comment]
Looks good to me. First time doing PR for this repo, not sure if I could click the "merge pull request" button, and I don't see an "approve" button here @digimaun

**Comment by WilliamBerryiii** [comment]
@digimaun do you want to do a review here too?

**Comment by WilliamBerryiii** [comment]
I'll keep going on a file by file basis and will keep adding to this PR unless there is good reason to mearge immediately.

**Comment by digimaun** [comment]
Since there is ongoing work on this PR, we can wait to review until it is finished.

**Comment by WilliamBerryiii** [comment]
@digimaun There is not too much more.  Just want to get commands and custom files in order.

**Comment by WilliamBerryiii** [comment]
@digimaun Ok, this is all done.  Filing a few issues now to close out the last of the questions.  Thanks for the review here!

File: azext_iot/_help.py [hunk 1]
```diff
@@ -61,23 +67,25 @@
...
     examples:
     - name: Turn on edge capabilities for device
       text: >
-        az iot hub device-identity update -d mydevice -n myhub --set capabilities.iotEdge=true
+        az iot hub device-identity update -d [Device ID] -n [IoTHub Name] 
+        --set capabilities.iotEdge=true
     - name: Disable device status
       text: >
-        az iot hub device-identity update -d mydevice -n myhub --set status=disabled
+        az iot hub device-identity update -d [Devie ID] -n [IoTHub Name] --set status=disabled
```
Comment (digimaun) [inline azext_iot/_help.py]: Small typo here.

File: azext_iot/_help.py [hunk 2]
```diff
@@ -61,23 +67,25 @@
...
     - name: In one command
       text: >
-        az iot hub device-identity update -d mydevice -n myhub --set status=disabled capabilities.iotEdge=true
+        az iot hub device-identity update -d [Devie ID] -n [IoTHub Name]
```
Comment (digimaun) [inline azext_iot/_help.py]: Small typo here.

File: azext_iot/_help.py [hunk 3]
```diff
@@ -29,23 +32,26 @@
     examples:
     - name: Create an edge enabled IoT device with default authorization (shared private key).
       text: >
-        az iot hub device-identity create -n myhub -d mydevice -ee
+        az iot hub device-identity create -n [IoTHub Name] -d [Device ID] -ee
     - name: Create an IoT device with self-signed certificate authorization,
             generate a cert valid for 10 days then use its thumbprint.
       text: >
-        az iot hub device-identity create -n myhub -d mydevice -am x509_thumbprint --valid-days 10
+        az iot hub device-identity create -n [IoTHub Name] -d [Device ID]
```
Comment (digimaun) [inline azext_iot/_help.py]: For understanding, what is your general strategy here, IoT Hub and Device Id's are obvious code replacement values, but you skip some other potential variables (like --output-dir and --valid-days)

File: azext_iot/_help.py [hunk 4]
```diff
@@ -29,23 +32,26 @@
...
             generate a cert valid for 10 days then use its thumbprint.
       text: >
-        az iot hub device-identity create -n myhub -d mydevice -am x509_thumbprint --valid-days 10
+        az iot hub device-identity create -n [IoTHub Name] -d [Device ID] 
+        -am x509_thumbprint --valid-days 10
     - name: Create an IoT device with self-signed certificate authorization,
             generate a cert of default expiration (365 days) and output to target directory.
       text: >
-        az iot hub device-identity create -n myhub -d mydevice -am x509_thumbprint --output-dir /path/to/output
+        az iot hub device-identity create -n [IoTHub Name] -d [Device ID] -am x509_thumbprint
```
Comment (digimaun) [inline azext_iot/_help.py]: For understanding, what is your general strategy here, IoT Hub and Device Id's are obvious code replacement values, but you skip some other potential variables (like --output-dir and --valid-days)

File: azext_iot/_help.py [hunk 5]
```diff
@@ -352,129 +366,141 @@
 
 helps['iot dps enrollment'] = """
     type: group
-    short-summary: Manage enrollments in an Azure IoT Hub device provisioning service
+    short-summary: Manage enrollments in an Azure IoT Hub Device Provisioning Service.
```
Comment (digimaun) [inline azext_iot/_help.py]: @YingXue  Can you validate the changes in character casing is OK for DPS? We have discussed rules for casing in the past for DPS.

File: azext_iot/_help.py (hunk 4 above)
Comment (digimaun) [reply azext_iot/_help.py]: (same applies to other commands)

File: azext_iot/_params.py [hunk 6]
```diff
@@ -4,6 +4,10 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
+"""
+CLI parameter definations.
```
Comment (digimaun) [inline azext_iot/_params.py]: Small typo here.

File: azext_iot/common/sas_token_auth.py [hunk 7]
```diff
@@ -4,6 +4,11 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
+"""
+sas_token_auth: Module containing Shared Access Signiture token class.
```
Comment (digimaun) [inline azext_iot/common/sas_token_auth.py]: Small typo 'Signiture'.

File: azext_iot/common/shared.py [hunk 8]
```diff
@@ -17,36 +26,80 @@ class SdkType(Enum):
...
+    Shared private key
+    """
     primary = 'primary'
     secondary = 'secondary'
 
 
+# pylint: disable=too-few-public-methods
 class AttestationType(Enum):
+    """
+    Type of atestation (TMP or certificate based)
```
Comment (digimaun) [inline azext_iot/common/shared.py]: Looking for uniformity on periods '.' at the end of these comment blocks. I don't have a strong opinion on whether or not it should be there, just the same should apply across the board.

File: azext_iot/common/shared.py [hunk 9]
```diff
@@ -17,36 +26,80 @@ class SdkType(Enum):
...
+    Resource status.
+    """
     disabled = 'disabled'
     enabled = 'enabled'
 
 
+# pylint: disable=too-few-public-methods
 class SettleType(Enum):
+    """
+    Result type of operation.
```
Comment (digimaun) [inline azext_iot/common/shared.py]: This enum defines how to settle a C2D message.

File: azext_iot/custom_sdk/custom_api.py [hunk 10]
```diff
@@ -143,10 +222,27 @@ def post_file_notification(
...
+            storage_endpoint (str): target url of container to post file to.
+            content (object): the content to post to the storage endpoint
+            content_type (dict): the IANA Media Type of the content.
+            raw (bool): return the result as a ClientRawResponse.
+            operation_config (): any msrest.service_client configuration overrides.
+
+        Raises:
+            ErrorDetailsException: when http response is not 200 or 201.
+
+        ToDo:
```
Comment (digimaun) [inline azext_iot/custom_sdk/custom_api.py]: thanks

File: azext_iot/_params.py [hunk 11]
```diff
@@ -23,143 +27,165 @@
...
-
-    with self.argument_context('iot dps registration') as c:
-        c.argument('registration_id', help='ID of device registration')
-
-    with self.argument_context('iot dps registration list') as c:
-        c.argument('enrollment_id', help='ID of enrollment group')
+    """
+    Load CLI Args for Knack parser
+    """
+    with self.argument_context('iot') as context:
```
Comment (digimaun) [inline azext_iot/_params.py]: Assuming no changes to argument definitions.

File: azext_iot/_help.py (hunk 4 above)
Comment (WilliamBerryiii) [reply azext_iot/_help.py]: They theory is that without replacement of the value the command may fail in a non-obvious way; e.g. 403, 404,  405, etc. With the "/path/to/output" you will end up with a file system access failure or completion of the request.  From what I have seen, "most" MSFT public docs follow this principle.

File: azext_iot/_help.py (hunk 5 above)
Comment (WilliamBerryiii) [reply azext_iot/_help.py]: The file used a mix of casing so I went with the proper noun for the service ... if that makes sense.

File: azext_iot/_help.py (hunk 1 above)
Comment (WilliamBerryiii) [reply azext_iot/_help.py]: fixed in commit: https://github.com/Azure/azure-iot-cli-extension/pull/4/commits/9647fc683a6f35fb880a1bcfdea8e2fefa19391b

File: azext_iot/_help.py (hunk 2 above)
Comment (WilliamBerryiii) [reply azext_iot/_help.py]: fixed in commit: https://github.com/Azure/azure-iot-cli-extension/pull/4/commits/9647fc683a6f35fb880a1bcfdea8e2fefa19391b

File: azext_iot/_params.py (hunk 6 above)
Comment (WilliamBerryiii) [reply azext_iot/_params.py]: fixed in commit: https://github.com/Azure/azure-iot-cli-extension/pull/4/commits/ab5b4b6c8effbd8bd7253254ca28e8aaedca1d07

File: azext_iot/common/sas_token_auth.py (hunk 7 above)
Comment (WilliamBerryiii) [reply azext_iot/common/sas_token_auth.py]: fixed in commit: https://github.com/Azure/azure-iot-cli-extension/pull/4/commits/10de7bb46355c2c7be79c877a6d80a2eee20d536

File: azext_iot/common/shared.py (hunk 8 above)
Comment (WilliamBerryiii) [reply azext_iot/common/shared.py]: GAH!  I missed one.  Fixed in commit: https://github.com/Azure/azure-iot-cli-extension/pull/4/commits/087bb44c55bb60ddbbac8fe1917adf42827a6a2b

File: azext_iot/common/shared.py (hunk 9 above)
Comment (WilliamBerryiii) [reply azext_iot/common/shared.py]: Updated in commit: https://github.com/Azure/azure-iot-cli-extension/pull/4/commits/087bb44c55bb60ddbbac8fe1917adf42827a6a2b

File: azext_iot/_params.py (hunk 11 above)
Comment (WilliamBerryiii) [reply azext_iot/_params.py]: Correct, just made `as` local more explicit.  I've historically only used single letters within the context of a closure, lambda or list comprehension as there is a narrowed scope.  For blocks of code like this, I prefer from a readability standpoint, more explicit names that are less likely to result in a naming conflict during editing.

**Comment by WilliamBerryiii** [comment]
Review commits pushed!  Thanks for the once-over @digimaun!

File: azext_iot/_help.py (hunk 5 above)
Comment (YingXue) [reply azext_iot/_help.py]: Sorry, I missed this comment, but the original case is following the guidelines. Anything that starts with an article (an or the in this case) should be lower case.
But seems like the extension if following different rules from Azure CLI Main, I am OK with this change for the consistency.

**Pull request merged**